### PR TITLE
merge 2.3 into 2.4

### PR DESCRIPTION
**Pull request by ronso0**
some test are failing:
```
LibraryFileWithMetadataSyncStatusUndefinedTest.saveTrackMetadataWithSyncEnabled (Failed)
LibraryFileWithoutMetadataSyncStatusUndefinedTest.saveTrackMetadataWithSyncEnabled (Failed)
```

### CODE REVIEW

**Comment by daschuer** [comment]
This is caused by this fix: 
https://github.com/mixxxdj/mixxx/pull/11127/commits/e8e1f0ad08cb913212f42c8803f5fe0b1403cafe
I will have a look.

File: src/sources/soundsourceproxy.cpp [hunk 1]
```diff
@@ -395,6 +395,7 @@ SoundSourceProxy::exportTrackMetadataBeforeSaving(
                 kLogger.warning()
                         << "Failed to update stream info from audio "
                            "source before exporting metadata";
+                return ExportTrackMetadataResult::Skipped;
```
Comment (daschuer) [inline src/sources/soundsourceproxy.cpp:398]: We need here adopt the change from: 
https://github.com/mixxxdj/mixxx/commit/a2b17efed1172810d6d2abcc8de646e0122c69cb
```suggestion
                return ExportTrackMetadataResult::Failed;
```

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/sources/soundsourceproxy.cpp (hunk 1 above)
Comment (ronso0) [reply src/sources/soundsourceproxy.cpp]: Thanks, tests pass now (locally)

**Comment by ronso0** [comment]
Ready!

**Review by daschuer — APPROVED** [review]
Thank you.

**Pull request merged**